### DISCUSSION TITLE
How kubegreen is different with Autoscalers(vpa/hpa)?

### DISCUSSION THREAD

**Discussion by foxutech** [kube-green/kube-green#319] [Q&A]
We have HPA, KEDA and VPA configured for some of our resources, how we can use kube-green along with this? How to by-pass it? what is the additional advantage compared to autoscaler.

**Comment by davidebianchi** [comment]
Hi @foxutech! The advantage of kube-green is to turn off the workload in an easy way, at fixed hours.
At the moment, I have not tried it with KEDA or VPA. I suppose that with VPA it is not a problem, since VPA should change only the resources and not the number of replicas. For KEDA, probably it's not necessary to use both of them since it should support the scale to 0 feature (I don't know if it supports also a fxed hour turn off of the workload).

For the HPA, they works well together since HPA does not touch deployment with 0 replicas. So, when workload are stopped, they remain set to 0 replicas.
